src/download.c

```diff
@@ -452,6 +452,9 @@ static int perform_curl_io_and_complete(int count, bool bounded)
 	return 0;
 }
 
+/* Fixed window added to hysteresis upper bound before it is enforced. */
+#define BUFFER 5
```

Comment:
If the upper bound is bypassed, are we guaranteed to run out of file descriptors? Or is this 5 a best guess at how much room we have before we run out?